mjit.c

```diff
@@ -108,9 +108,14 @@ mjit_gc_exit_hook(void)
     if (!mjit_enabled)
         return;
     CRITICAL_SECTION_START(4, "mjit_gc_exit_hook");
-    in_gc = false;
-    verbose(4, "Sending wakeup signal to workers after GC");
-    rb_native_cond_broadcast(&mjit_gc_wakeup);
+    in_gc--;
+    if (in_gc < 0) { // Don't allow underflow
```

Comment:
Do we need this? I thought we could just assert `in_gc >= 0` instead.

Reply:
I added that assertion, but for some reason I got an assertion failure when the build runs.  It seems like something is calling `exit` too many times. 😅

Reply:
https://github.com/ruby/ruby/pull/3687 Travis is stuck, but other CIs have passed 🤔 

How did you reproduce the issue?